libraries/chain/database.cpp

```diff
@@ -3809,6 +3809,19 @@ void database::apply_hardfork( uint32_t hardfork )
       case STEEMIT_HARDFORK_0_18:
          break;
       case STEEMIT_HARDFORK_0_19:
+         {
+            const auto& delegation_idx = get_index< vesting_delegation_index, by_id >();
+            auto itr = delegation_idx.begin();
+
+            while( itr != delegation_idx.end() )
```

Comment:
We do not track that index. This is the safe way to implement the fix. The 0 delegation objects should have no effect on consensus. After the hardfork we can change the 0.17 behavior to match 0.19 and verify that not creating 0 delegation objects has no effect. In which case we can delete this logic entirely.